### DISCUSSION TITLE
Aula 03 - Criando a Estrutura do Projeto

### DISCUSSION THREAD

**Discussion by rikemorais** [balta-io/7196#27] [Perguntas e Respostas] [answered]
Boa noite!

Quando eu executei o:

`dotnet new webapi`

Para o Todo.Domain.Api eu recebi o seguinte erro:

`error NU1101: Não é possível encontrar o pacote Swashbuckle.AspNetCore. Não existe nenhum pacote com esta ID nas origens: Microsoft Visual Studio Offline Packages`

Pesquisei o erro na [Documentação](https://docs.microsoft.com/pt-br/nuget/reference/errors-and-warnings/nu1101), mas, não ficou claro o que aconteceu. Poderia me dizer o que pode ter sido?

Desde já, agradeço.

**Comment by rikemorais** [comment]
Com a pasta .Domain.Tests o erro foi diferente:

`
dotnet new mstest
O modelo "MSTest Test Project" foi criado com êxito.

Processando ações pós-criação...
Executando 'dotnet restore' em C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj...
  Determinando os projetos a serem restaurados...
C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj : error **NU1102: Não é possível encontrar o pacote coverlet.collector com a versão (>= 3.1.0)**
C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj : **error NU1102:   - Encontradas 1 versões em Microsoft Visual Studio Offline Packages [Versão mais próxima: 3.0.3]**
C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj : **error NU1102: Não é possível encontrar o pacote NuGet.Frameworks com a versão (>= 5.0.0)**
C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj : **error NU1102:   - Encontradas 1 versões em Microsoft Visual Studio Offline Packages [Versão mais próxima: 4.6.4]**
  Falha ao restaurar C:\dev\Gantting\Gantting.Domain.Tests\Gantting.Domain.Tests.csproj (em 529 ms).
Falha na restauração.
Falha na postagem.
Instruções manuais: Run 'dotnet restore'
`

Estou chamando meu projeto de Gantting. Estou pesquisando aqui o que pode ser. Pelo que o erro descreve, devo estar com essas dependências desatualizadas. Vou ver como atualizo. Se alguém puder ajudar, eu agradeço. Tmj

**Comment by rikemorais** [comment, marked as answer]
Continuei pesquisando e tentando. Instalei a versão 3.1 do .NET Framework Core e agora compila com 2 avisos sobre o `MSTest.TestFrame`. Se possível, gostaria de entender o que aconteceu nesse caso:

`NU1603: Gantting.Domain.Tests depende de MSTest.TestFramework (>= 2.1.0), mas MSTest.TestFramework 2.1.0 não foi encontrado. A melhor correspondência aproximada de MSTest.TestFramework 2.2.7 foi resolvida.`

**Reply by andrebaltieri** [reply]
Bom dia, @RikeMorais como vai?
Este erro já ocorreu comigo e no meu caso, por algum motivo quando o .NET intalou ele não registrou o Nuget como source para os pacotes... ai ele não achava nada..

A solução é a seguinte, executa o comando `dotnet nuget list source` e veja se tem algum source de pacotes na sua máquina.
Se não houver, adiciona o source do nuget `dotnet nuget add source https://api.nuget.org/v3/index.json -n nuget.org`

**Reply by rikemorais** [reply]
Master, muito obrigado. Você é o melhor. Resolvidíssimo. Uma honra ser seu aluno. Tmj.

**Reply by andrebaltieri** [reply]
<3
